README.md

```diff
@@ -29,6 +29,61 @@ PS: Latest Build Status is against `main` branch.
 | iOS | [![CocoaPods](https://img.shields.io/cocoapods/v/AdaptiveCards.svg)](https://cocoapods.org/pods/AdaptiveCards) | [Source](https://github.com/Microsoft/AdaptiveCards/tree/main/source/ios) | [Docs](https://docs.microsoft.com/en-us/adaptive-cards/display/libraries/ios) |  ![Build status](https://img.shields.io/azure-devops/build/Microsoft/56cf629e-8f3a-4412-acbc-bf69366c552c/37917/main.svg) |
 | Card Designer | [![npm install](https://img.shields.io/npm/v/adaptivecards-designer.svg)](https://www.npmjs.com/package/adaptivecards-designer) | [Source](https://github.com/Microsoft/AdaptiveCards/tree/main/source/nodejs/adaptivecards-designer)| [Docs](https://www.npmjs.com/package/adaptivecards-designer) | ![Build Status](https://img.shields.io/azure-devops/build/Microsoft/56cf629e-8f3a-4412-acbc-bf69366c552c/20564/main.svg) |
 
+## Code format
+
+We require the C++ code inside this project to follow the clang-format. If you change them, please make sure your changed files are formatted correctly.
+
+Make sure clang-format version 12.0.0 and above version are used.
+
+### IDE integration
+ClangFormat describes a set of tools that are built on top of LibFormat. It can support your workflow in a variety of ways including a standalone tool and editor integrations. For details, refer to https://clang.llvm.org/docs/ClangFormat.html
+
+### Format with scripts
+Two scripts are provided to help you format files.
+- Windows only user: use FormatSource.ps1. This script use clang-format.exe which is built into Visual Studio by default.
+
+	Execute below command in the root folder of the project
+
+	```
+	PowerShell.exe -ExecutionPolicy Bypass scripts\FormatSource.ps1 -ModifiedOnly $False
+	```
+
+If it's the first time to run the script, make sure clang-format version 12.0.0 or above in the output. Otherwise you may need to upgrade Visual Studio or use your own clang-format binaries.
+```
+[clang-format] Version is:
+clang-format version 12.0.0
+```
+
+- Both Windows and MAC user: Use clang-format npmjs package
+
+	Execute below command in source/nodejs
+
+	```
+	npm run format
+	``` 
```

Comment:
This will be used by the pipeline. 
In my testing. clang-format-12 is enough for cpp files, but for objective-c and java, clangformat 13 or above is required.

Reply:
I will keep it as it is.